### PR TITLE
Added setup instructions for Github Codespaces

### PR DESCRIPTION
**Pull request by utsab**
Checklist:

- [x ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x ] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

### CODE REVIEW

**Comment by lloydchang** [comment]
LGTM

I ran it via the following:
```
@lloydchang ➜ /workspaces/freeCodeCamp-classroom (main) $ chmod 755 codespace-instructions.sh 
@lloydchang ➜ /workspaces/freeCodeCamp-classroom (main) $ ./codespace-instructions.sh 
<------------ SETUP INSTRUCTIONS ------------>

Almost done! Just need to set up a NextAuth Provider and run the app.

1.) For dev environments, we have enabled use of the Github Provider. Please follow their guide:
https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app

The mock user data URL is:
https://lloydchang-jubilant-goldfish-5gpv5vww5g297x-3001.preview.app.github.dev/data

The homepage is:
https://lloydchang-jubilant-goldfish-5gpv5vww5g297x-3000.preview.app.github.dev

The callback url is:
https://lloydchang-jubilant-goldfish-5gpv5vww5g297x-3000.preview.app.github.dev/api/auth/callback/github

2.) Set NEXTAUTH_URL in .env to https://lloydchang-jubilant-goldfish-5gpv5vww5g297x-3000.preview.app.github.dev

3.) To setup GITHUB_ID and GITHUB_SECRET after setting up the OAuth app:
-Your Client ID is shown in the OAuth app page. Copy and paste to GITHUB_ID.
-Next generate a new Client Secret. Then copy and paste to GITHUB_SECRET.

Additional Steps to take After Finishing Gitpod Setup:

1.) Run: npm run develop

2.) Check to see if the website is functional and you can sign in and authorize yourself using your GitHub profile.

3.) After successfully signing in, sign out and open a new terminal.

4.) Run: npx prisma studio

5.) In Prisma Studio, go to User. Under the “roles” column, enter TEACHER or ADMIN. And click on the “Save Changes” button.

6.) Run npm run mock-fcc-data

7.) Go to the ports tab and set the visibility to public for the 3000 and 3001 ports.

Happy coding!
@lloydchang ➜ /workspaces/freeCodeCamp-classroom (main) $
```

**Comment by lloydchang** [comment]
Separately:

I don't see a `.devcontainer` folder with a `devcontainer.json` file in this repository.

Are you using [GitHub Codespaces' default container configuration](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers#using-the-default-dev-container-configuration)?

If yes, did you encounter an error after running `npm ci`?

I submitted a fix via https://github.com/freeCodeCamp/classroom/pull/350

**Comment by lloydchang** [comment]
From https://github.com/freeCodeCamp/classroom/pull/349#issuecomment-1624222418

---
> **Good news!**
---
> **I tested this https://github.com/freeCodeCamp/classroom/pull/349 — GitHub Codespaces and Gitpod are at parity; they are equal.**
---
> **Please see the screenshots below where the GitHub Codespaces screenshot looks the same as the Gitpod screenshot:**
---
> **GitHub Codespaces Screen Shot 2023-07-06 at 12 41 20 PM.png**
---
> 
> GitHub Codespaces Screen Shot 2023-07-06 at 12 41 20 PM
> 
---
> **Gitpod Screen Shot 2023-07-06 at 10.20.34 AM.png**
---
> 
> Gitpod Screen Shot 2023-07-06 at 10 20 34 AM
>

**Review by GuillermoFloresV — APPROVED** [review]

File: codespace-instructions.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+echo The callback url is:
+echo https://${CODESPACE_NAME}-3000.preview.app.github.dev/api/auth/callback/github
+echo
+echo 2.\) Set NEXTAUTH_URL in .env to https://${CODESPACE_NAME}-3000.preview.app.github.dev
+echo
+echo 3.\) To setup GITHUB_ID and GITHUB_SECRET after setting up the OAuth app: 
+echo      -Your Client ID is shown in the OAuth app page. Copy and paste to GITHUB_ID. 
+echo      -Next generate a new Client Secret. Then copy and paste to GITHUB_SECRET.
+echo
+echo Additional Steps to take After Finishing Gitpod Setup:
```
Comment (lloydchang) [inline codespace-instructions.sh:25]:
```Suggestion
echo Additional Steps to take After Finishing Codespaces Setup:
```

Hi @utsab and @GuillermoFloresV 

There was a copy-and-paste minor typo:

Via https://github.com/freeCodeCamp/classroom/pull/349, I fixed it at https://github.com/freeCodeCamp/classroom/pull/349/files#diff-554f97261a97badb0ab3b9ed138c0408923adee822657cc18cd7ae86e5484b2bR25

**Pull request merged**